cli/azd/pkg/tools/azcli/keyvault.go

```diff
@@ -0,0 +1,133 @@
+package azcli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault"
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+	"github.com/azure/azure-dev/cli/azd/pkg/identity"
+)
+
+type AzCliKeyVault struct {
```

Comment:
I'm wondering if we should continue to use these types or just move over to the structures that the go SDK exports?  I'm wondering if the ceremony ends up being worth the cost of the decoupling.

If we do keep these, maybe we should consider dropping the `json` serialization type tags from them?  These were more or less model types from the CLI and the JSON tags were there so we could use the normal go marshaller to parse the return values from the CLI.

Interested in your thoughts.

Reply:
I'm in agreement but would like to take this in phases...

1. Update current `azcli` implementation to use go modules with minimal interface changes
2. Refactor / break apart `azcli` and use better types.